### PR TITLE
Convert Workflow from HCL to YAML

### PR DESCRIPTION
**Pull request by stefanzweifel**
Attempt to Migrate the existing HCL Workflow to the YAML format.
https://help.github.com/en/articles/migrating-github-actions-from-hcl-syntax-to-yaml-syntax

### CODE REVIEW

**Comment by stefanzweifel** [comment]
Look like I'm not yet in the new YAML beta 🤷‍♂

**Comment by joelwmale** [comment]
I have tested your workflow @stefanzweifel as I have YAML beta access. It fails every time at the composer install step:

```
Your requirements could not be resolved to an installable set of packages.
##[error]Docker run failed with exit code 2
```

I'm confident it's not a me issue as I have 0 issues doing fresh installs with this project.

**Comment by stefanzweifel** [comment]
The old HCL-workflow has been deleted.

I've decided to split the existing workflow into separate files, for easier maintenance. There are now the following workflows:

- `test`: Run the PHPUnit Testsuite on each push
- `insights`: Run PHP Insights on each push and display the results in the Action's log
- `format_php`: Run PHP CS Fixer on each push within a Pull Request. If changes have been made, push those changes back to the repository.

**Comment by TomasVotruba** [comment]
Thanks for a great post Stefan. It gave me quick insight into the topic, really well written :+1:

File: .github/workflows/format_php.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    - name: Run php-cs-fixer
+      uses: docker://oskarstark/php-cs-fixer-ga
+
+    - name: Commit changed files
+      uses: stefanzweifel/git-auto-commit-action@v2.1.0
+      with:
+        commit_message: Apply php-cs-fixer changes
+        branch: ${{ github.head_ref }}
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (TomasVotruba) [inline .github/workflows/format_php.yml]: Does this really work?
It looks amazing

Comment (stefanzweifel) [reply .github/workflows/format_php.yml]: Yes this does work :)

We use this or a very similar workflow in a lot of projects at my current job.
It seems I haven't made an example of actually running the workflow in a PR in this repository.

You can find a working example/demo in the demo project I use for the mentioned GitHub Action here: https://github.com/stefanzweifel/git-auto-commit-action-demo-app/pull/4

(Sorry that there are a lot of "noise" commits, but I think you will see what's possible. I just recently made that repo public 😅.)

Comment (TomasVotruba) [reply .github/workflows/format_php.yml]: Thanks a lot, I gotta try it now :) so futuristic

**Pull request merged**